include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -1242,6 +1558,75 @@ DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::
 
 
 
+template <typename DoFHandlerType, int patch_dim, int patch_space_dim>
+template <class VectorType>
+void
+DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::add_mg_data_vector(
+  const DoFHandlerType &           dof_handler,
+  const MGLevelObject<VectorType> &data,
+  const std::string &              name)
+{
+  // forward the call to the vector version:
+  std::vector<std::string> names(1, name);
+  add_mg_data_vector(dof_handler, data, names);
+}
+
+
+
+template <typename DoFHandlerType, int patch_dim, int patch_space_dim>
+template <class VectorType>
+void
+DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::add_mg_data_vector(
+  const DoFHandlerType &           dof_handler,
+  const MGLevelObject<VectorType> &data,
+  const std::vector<std::string> & names,
+  const std::vector<DataComponentInterpretation::DataComponentInterpretation>
+    &data_component_interpretation_)
+{
+  if (triangulation == nullptr)
```

Comment:
How does this function differ from the non-MG function? It does a lot of stuff, but I think only the last line is actually using a different type, right? I wonder whether we could just use the usual function and have some sort of dispatch mechanism that picks the right type only in the last line, depending on the vector kind -- and then we could just use `add_data_vector()` (without distinguishing between mg and non-mg versions in the name of the function).

Reply:
I was thinking about doing this originally, but this would add to the existing 6(?) overloads, which could lead to confusion. This might be a matter of taste, but I find it much harder to parse error messages when there are many overloads. 
I also dislike the automatic cell vs dof detection that fails if you have a DGQ0. :-)